apps/researcher/src/lib/community.ts

```diff
@@ -28,8 +31,58 @@ export async function getMemberships(
   });
 }
 
-export function getAllCommunities(): Promise<Community[]> {
-  return clerkClient.organizations.getOrganizationList();
+export enum CommunitySortBy {
+  NameAsc = 'nameAsc',
+  NameDesc = 'nameDesc',
+  CreatedAtDesc = 'createdAtDesc',
+  MembershipCountDesc = 'membershipCountDesc',
+}
+
+export const defaultCommunitySortBy: CommunitySortBy =
+  CommunitySortBy.CreatedAtDesc;
+
+export function sortCommunities(
+  communities: Community[],
+  sortBy: CommunitySortBy
+) {
+  return [...communities].sort((a, b) => {
+    if (sortBy === CommunitySortBy.NameAsc) {
+      return a.name.localeCompare(b.name);
+    } else if (sortBy === CommunitySortBy.NameDesc) {
+      return b.name.localeCompare(a.name);
+    } else if (sortBy === CommunitySortBy.CreatedAtDesc) {
+      return b.createdAt - a.createdAt;
+    } else {
+      return 0;
+    }
+  });
+}
+
+interface GetAllCommunitiesProps {
+  query?: string;
+  sortBy?: CommunitySortBy;
+  limit?: number;
+  offset?: number;
+}
+
+export async function getAllCommunities({
```

Comment:
Minor: could we rename the function to `getCommunities()` (which implies 'all')? Or is there a difference between getting 'communities' and 'all communities'?